paddle/fluid/operators/transfer_layout_op.h

```diff
@@ -54,8 +56,18 @@ class TransferLayoutFunctor {
     out_tensor.set_layout(out_layout);
 
 #ifdef PADDLE_WITH_MKLDNN
+    // NOTE(zhiqiu): to handle the special case in ApplyDataTransform() in
+    // data_transfer.cc
     auto in_layout = static_cast<DataLayout>(src_layout_);
+    auto *tensor_out = out_->GetMutable<framework::LoDTensor>();
     VLOG(4) << in_layout << "->" << out_layout << " " << in_tensor.layout();
+    if (!in_tensor.IsInitialized() && in_layout == DataLayout::kMKLDNN &&
+        out_layout == DataLayout::kNHWC) {
```

Comment:
Recently DataLayout was extended with 5D NHWC layout e.g. kNDHWC for which also MatchShapeToLayout() should be called

Reply:
Ok, I think this can be refined in the future, here just follow the same logic in executor.cc

Reply:
ok